src/widgetastic_patternfly/__init__.py

```diff
@@ -254,7 +255,7 @@ class NavDropdown(Widget, ClickableMixin):
 
     def __init__(self, parent, id=None, logger=None):
         """id is optional to allow for easy subclass change of ROOT"""
-        Widget.__init__(self, parent, logger=logger)
+        GenericLocatorWidget.__init__(self, parent, locator=None, logger=logger)
```

Comment:
I think a better fix would be just to modify the `__repr__` method for this class, so that it uses the `id` attribute, which is the attribute used to instantiate `NavDropdown` and `SettingsNavDropdown` instances. The following patch works in my testing:

```
$ git diff
diff --git a/src/widgetastic_patternfly/__init__.py b/src/widgetastic_patternfly/__init__.py
index b8432fe..ebd6eea 100644
--- a/src/widgetastic_patternfly/__init__.py
+++ b/src/widgetastic_patternfly/__init__.py
@@ -345,7 +345,7 @@ class NavDropdown(Widget, ClickableMixin):
         self.browser.click('./ul/li[normalize-space(.)={}]'.format(quote(item)), parent=self)
 
     def __repr__(self):
-        return '{}({!r})'.format(type(self).__name__, self.locator)
+        return '{}(id={!r})'.format(type(self).__name__, self.id)
 
 
 class BootstrapNav(Widget):
```

Before the patch:

```
-> view = navigate_to(appliance.server, 'LoggedIn')
(Pdb) n
[...]
(Pdb) view.help
*** AttributeError: 'NavDropdown' object has no attribute 'locator'

(Pdb) view.settings
*** AttributeError: 'SettingsNavDropdown' object has no attribute 'locator'
```

After:

```
-> view = navigate_to(appliance.server, 'LoggedIn')
(Pdb) n

(Pdb) view.settings
SettingsNavDropdown(id='dropdownMenu2')

(Pdb) view.help
NavDropdown(id='help-menu')
```

Reply:
I think you are right. I did it as you suggested.